armi/materials/tests/test_materials.py

```diff
@@ -139,6 +144,60 @@ def test_findMaterial(self):
                 "Unobtanium", namespaceOrder=["armi.materials"]
             )
 
+    def __validateMaterialNamespace(self):
+        """Helper method to validate the material namespace a little."""
+        self.assertTrue(isinstance(_MATERIAL_NAMESPACE_ORDER, list))
+        self.assertGreater(len(_MATERIAL_NAMESPACE_ORDER), 0)
+        for nameSpace in _MATERIAL_NAMESPACE_ORDER:
+            self.assertTrue(isinstance(nameSpace, str))
+
+    @unittest.skipUnless(context.MPI_RANK == 0, "test only on root node")
+    def test_namespacing(self):
+        """Test loading materials with different material namespaces, to cover how they work.
+
+        .. test:: Material can be found in defined packages.
+            :id: T_ARMI_MAT_NAMESPACE1
+            :tests: R_ARMI_MAT_NAMESPACE
+
+        .. test:: Material namespaces register materials with an order of priority.
+            :id: T_ARMI_MAT_ORDER
+            :tests: R_ARMI_MAT_ORDER
+        """
+        # let's do a quick test of getting a material from the default namespace
+        setMaterialNamespaceOrder(["armi.materials"])
+        uo2 = materials.resolveMaterialClassByName(
+            # "Hafnium", namespaceOrder=["armi.materials"]
+            "UO2",
+            namespaceOrder=["armi.materials"],
+        )
+        self.assertGreater(uo2().density(500), 0)
+
+        # validate the default namespace in ARMI
+        self.__validateMaterialNamespace()
+
+        # show you can add a material namespace
+        newMats = "armi.utils.tests.test_densityTools"
+        setMaterialNamespaceOrder(["armi.materials", newMats])
+        self.__validateMaterialNamespace()
+
+        # show that adding a name material namespace provides access to new materials
+        testMatIgnoreFake = materials.resolveMaterialClassByName(
+            "TestMaterialIgnoreFake", namespaceOrder=["armi.materials", newMats]
+        )
+        for t in range(200, 600):
+            self.assertEqual(testMatIgnoreFake().density(t), 0)
+            self.assertEqual(testMatIgnoreFake().pseudoDensity(t), 0)
+
+        # show the ordering of the material namespaces works
+        # (The second material namespace would set the density of UO2 to zero.)
```

Comment:
I think your second comment would be a good to assertion to show.

Reply:
Okay, this and some other stuff was left over from a different version of the tests. I've cleaned a ton of this out.